pkg/apis/clusterregistry/types.go

```diff
@@ -114,12 +114,31 @@ type AuthProviderConfig struct {
 	// +optional
 	Name string
 
+	// Type contains type information about this auth provider. Clients of the
+	// cluster registry should use this field to differentiate between different
+	// kinds of authentication providers.
+	// +optional
+	Type AuthProviderType
+
 	// Config is a map of values that contains the information necessary for a
 	// client to determine how to authenticate to a Kubernetes API server.
 	// +optional
 	Config map[string]string
 }
 
+// AuthProviderType contains metadata about the auth provider. It should be used
+// by clients to differentiate between different kinds of auth providers, and so
```

Comment:
nit: `s/so//`?

Reply:
Good catch! I actually meant "to", not "so". Fixed.